tests/microos/toolbox.pm

```diff
@@ -0,0 +1,88 @@
+    # Display help
+    assert_script_run 'toolbox -h';
+
+    record_info 'Test',                    "Run toolbox without flags";
+    assert_script_run 'toolbox -r id',     timeout => 180;
+    validate_script_output 'podman ps -a', sub { m/toolbox-root/ };
+    assert_script_run 'podman rm toolbox-root';
+
+    record_info 'Test',                         "Run toolbox with a given tag";
+    assert_script_run 'toolbox -t test_tag id', timeout => 180;
+    validate_script_output 'podman ps -a',      sub { m/toolbox-root-test_tag/ };
+    assert_script_run 'podman rm toolbox-root-test_tag';
+
+    record_info 'Test', "Run toolbox with a given name";
+    assert_script_run('toolbox -c test_name id');
+    validate_script_output 'podman ps -a', sub { m/test_name/ };
+    assert_script_run 'podman rm test_name';
+
+
+    record_info 'Test', "Rootless toolbox as $user";
+    my $prefix = "runuser -l $user -c";
+    my $uid    = script_output "$prefix 'id -u'";
+    validate_script_output "$prefix 'toolbox -u id'", sub { m/uid=${uid}\(${user}\)/ }, timeout => 180;
+    die "$user shouldn't have access to /etc/passwd!" if (script_run("$prefix 'toolbox -u touch /etc/passwd'") == 0);
+    assert_script_run "$prefix \"podman rm toolbox-$user-user\"";
+
+    record_info 'Test',                                         "Rootfull toolbox as $user";
+    validate_script_output "$prefix 'toolbox -r id'",           sub { m/uid=0\(root\)/ };
+    assert_script_run "$prefix 'toolbox -r touch /etc/passwd'", fail_message => 'Root should have access to /etc/passwd!';
+    assert_script_run "podman rm toolbox-$user";
+
+
+    record_info 'Test', 'Pulling toolbox toolbox -u idimage from different registry';
+    # Switch default registries for openSUSE MicroOS and SLE Micro
+    if (is_sle_micro) {
+        assert_script_run 'echo -e "REGISTRY=registry.opensuse.org\nIMAGE=opensuse/toolbox" > ~/.toolboxrc';
+        validate_script_output 'toolbox -r cat /etc/os-release', sub { m/opensuse/ }, timeout => 180;
+    } else {
+        assert_script_run 'echo -e "REGISTRY=registry.suse.com\nIMAGE=suse/sle-micro/5.0/toolbox" > ~/.toolboxrc';
+        validate_script_output 'toolbox -r cat /etc/os-release', sub { m/sles/ }, timeout => 180;
+    }
+    assert_script_run 'podman rm toolbox-root';
+    assert_script_run 'rm ~/.toolboxrc';
+
+    record_info 'Test', 'Zypper tests';
+    assert_script_run 'toolbox create -r -c devel';
+    if (!validate_script_output 'toolbox list', sub { m/devel/ }, timeout => 180, proceed_on_failure => 1) {
+        record_info('ISSUE', 'https://github.com/kubic-project/microos-toolbox/issues/23');
+    }
+    script_run 'toolbox run -c devel -- zypper lr';    # this command will fail in SLE Micro toolbox as there are no repos
+    assert_script_run 'toolbox run -c devel -- zypper -n in python3' unless is_sle_micro;
+    assert_script_run 'podman rm devel';
+
+    # Leave a clean system
+    assert_script_run 'podman rm --all --volumes --force';
```

Comment:
I would move this into a function a call this function in `post_fail` / `post_run` hooks so we will clean not only when everything is fine